### PR TITLE
Update Liblouis to version 3.5.0

### LINKED ISSUE #7301: Disable pass1only for liblouis braille output
Note: This issue needs an external fix at the moment, liblouis/liblouis#133 . I opened this issue to document cases why using Liblouis multipass is desired.

### Steps to reproduce:
1. Set braille output table to Dutch (Netherlands)
2. Open notepad, and type: "1b3d5"

### Expected behavior:
Output is: ⠼⠁⠠⠃⠼⠉⠠⠙⠼⠑

### Actual behavior:
Output is: ⠼⠁⠃⠼⠉⠙⠼⠑
It is not clear how to read this. It is not possible to see where the numbers end and the letters start. In this example, you may be able to guess that there is a b and a d, not a 2 and a 4. However, In strings like 12cde67hij, you certainly won't be able to see where the letters start.

### Technical background
NVDA sends text to the liblouis braille translator using the pass1Only mode flag. From liblouis/liblouis#133:

> trace_translate , the general translation function in lou_translateString.c calls translatePass for pass2, 3 and 4. translatePass does currently not update the output positions.

Specifically, this leads to the problem that cursor routing is said to break in NVDA as the braille character to output position mapping is broken. I wasn't able to reproduce this using the example above, though.

CC @JosephSL @DKager

### LINKED ISSUE #7526: Danish contracted eight dot braille doesn't show capital characters
NVDA 2017.3 does not show capital characters when using the 8 dot contracted braille table.
This was the case until 2017.3 where the input table was added.

Here is how to reproduce

1. Select the Danish 8 dot contracted table under options, braille options.
2. Open Notepad and type:
Hello World
Now make a newline and type:
NVDA
Remember to type all characters as capitals when typing NVDA.
3. On the braille display it shows:
hello world!
n v d a 

So:
When a word contains of multiple capital letters the dot 7 is not used, instead there is inserted a space, and when words start with a capital letter this is not shown.

4. Now try retype the examples above but switch to the Danish uncontracted braille table.
You will now find that capital characters works again just fine.

Tested on Windows 7 with a Braille Sense U2 from HIMS.

### LINKED ISSUE #7693: Urgent: Please add option to enable multi-pass mode for Liblouis under settings/Braille
Currently, NVDA uses Liblouis in pass_1_only mode, presumably to ensure correct cursor positions. However, this seriously messes up Braille with all Liblouis tables that use multi-pass opcodes, such as Danish, Polish and probably other tables as well. se #7526 and #7301.

With the above option, the user could (at his or her own risk) enable multi-pass mode, until you feel it is safe to do this generally.

If you feel it is appropriate, you could add a warning stating that enabling this option might cause any number of catastrophic consequences to yourself, your guide dog and other loved ones, not to mention your cursor position, but please, make the option available.

### LINKED ISSUE #7847: Include braille tables for romanian language in NVDA's braille settings
### Steps to reproduce:
1. Connect a braille display with NVDA and write following:
Romanian: Haș
2. Try to read the words on the Braille Display
3. Go to the last letter of the romanian word and try to read the single character with eSpeak

### Expected behavior:
The letter ș as well as other language specıfıc signs should automatically be properly dısplayed on a braılle dısplay. ș is for example dots 2, 4, 6 (grade 1).

### Actual behavior:
On the Braille Display there is a code generated (for romanian letter for example 'x219') instead of displaying the letter itself.

### System configuration:
NVDA version:
2017.4

NVDA Installed or portable:
Both

Other information:

Windows version:
Windows 10 Build 16299.64

Name and version of other software in use when reproducing the issue:
All

### Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
Yes, 2017.3, 2017.1

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Supersedes #7822.
Follow-up of #7702
Closes #7301
Fixes #7693, and possibly other issues
Fixes #7526
Closes #7847

### Description of this pull request:
First and foremost, this updates the Liblouis braille translator to version 3.5, which is the most recent version. This includes a new Lithuanian 8 dot table, a Mongolian grade 2 table and a Ukrainian table.

In #7702, we introduced a hidden config parameter to toggle the use of Liblouis multipass vs. pass1only. In that pr, pass1only was still enabled, although you could disable it by changing a hidden config parameter. E.g.:
```import config; config.conf["braille"]["outputPass1Only"]=False```

This pr removes this flag altogether, as pass1only has been removed from liblouis.

This also adds two unit tests for the brailleTables module:

* A test to check whether all defined tables actually exist in the tables folder
* A test whether all renamed tables are part of the list of defined tables

### Testing performed
* Tested both Lithuanian tables.
* Tested the Ukrainian table.
* Tested Mongolian grade 2.

### Change log entry:
* New features
    + New braille translation tables: Lithuanian 8 dot, Ukrainian, Mongolian grade 2

* Changes
    + Updated liblouis braille translator to 3.5.0. (#7839)
    + The Lithuanian braille table has been renamed to Lithuanian 6 dot to avoid confusion with the new 8 dot table. (#7839)
    + The French (Canada) grade 1 and grade 2 tables have been removed. Instead, the French (unified) 6 dot computer braille and Grade 2 tables will be used respectively.

* Bug fixes
    + Fixed malformed braille output for several braille tables, including capital signs in 8 dot contracted Danish braille. (#7526, #7693)

* Changes for developers
    + The hidden boolean flag "outputPass1Only" in the braille section of the configuration has again been removed. Liblouis no longer supports pass 1 only output.

### CODE REVIEW

**Comment by josephsl** [comment]
Hi, let’s wait for 3.4.1 (next week) before incubating it. Thanks.

**Comment by leonardder** [comment]
[**@josephsl**](https://github.com/josephsl) commented on [8 dec. 2017 08:22 CET](https://github.com/nvaccess/nvda/pull/7839#issuecomment-350192303 "2017-12-08T07:22:25Z - Replied by Github Reply Comments"):
> Hi, let’s wait for 3.4.1 (next week) before incubating it. Thanks.

What is your source regarding this potential 3.4.1 release? There is no milestone for this release.

**Comment by josephsl** [comment]
Hi, Liblouis mailing list. Thanks.

**Comment by leonardder** [comment]
[Here](https://www.freelists.org/post/liblouis-liblouisxml/Status-bug-fix-release) is the most recent update for the bugfix release (3.4.1).

**Comment by leonardder** [comment]
@josephsl: Curious to know your opinion about the current state of Liblouis. It seems that development is heavily delayed.

I don't think it is a good idea to force Liblouis 3.4.1 upon users of NVDA 2017.1. I could base this pr on Liblouis 3.4.1 pre 1, but I don't think that doesn't make any sense either.

**Comment by josephsl** [comment]
Hi, agreed. I motion that we come back to it once 2018.1 hits the air. Thanks.

**Comment by bertfrees** [comment]
The planned 3.4.1 release was cancelled because it's almost time for the next regular release (3 weeks). However it would still be useful if you guys could test the [v3.4.1-pre1 tag](https://www.freelists.org/post/liblouis-liblouisxml/Status-bug-fix-release,1).

**Comment by leonardder** [comment]
[**@bertfrees**](https://github.com/bertfrees) commented on [12 feb. 2018 12:27 CET](https://github.com/nvaccess/nvda/pull/7839#issuecomment-364896130 "2018-02-12T11:27:32Z - Replied by Github Reply Comments"):
> The planned 3.4.1 release was cancelled because it's almost time for the next regular release (3 weeks). However it would still be useful if you guys could test the [v3.4.1-pre1 tag](https://www.freelists.org/post/liblouis-liblouisxml/Status-bug-fix-release,1).

I agree this could be beneficial.

@michaeldcurran: How about incubating this pr to allow some testing to be done, and than update the pr to 3.5 when that is out?

File: readme.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ For reference, the following dependencies are included in Git submodules:
 * [Sonic](https://github.com/waywardgeek/sonic), commit 4f8c1d11
 * [IAccessible2](http://www.linuxfoundation.org/collaborate/workgroups/accessibility/iaccessible2), version 1.3
 * [ConfigObj](http://www.voidspace.org.uk/python/configobj.html), version 4.6.0
-* [liblouis](http://www.liblouis.org/), version 3.3.0
+* [liblouis](http://www.liblouis.org/), version 3.4.1
```
Comment (feerrenrut) [inline readme.md]: Might want to update this to 3.4.1Pre1

**Comment by leonardder** [comment]
Liblouis 3.5 will most certainly drop support for pass1only completely, see liblouis/liblouis#511. cc @BueVest

@michaeldcurran: with this in mind, we could consider removing the outputPass1Only what's new entry for 2018.1, since it probably will only survive that version. This proposal does not conflict with the translation freeze.

**Comment by Andre9642** [comment]
@leonardder The french tables `fr-fr-g1.utb', `fr-fr-g2.ctb', `fr-ca-g1.utb' and `fr-ca-g2.ctb' have been removed.
Could you remove the necessary in 'brailleTables.py'? Among others:

```
addTable("fr-ca-g1.utb", _("French (Canada) grade 1"))
addTable("Fr-Ca-g2.ctb", _("French (Canada) grade 2"), contracted=True)
```

Thanks

**Comment by leonardder** [comment]
@michaeldcurran: This is now fully based on Liblouis 3.5, so your review would be greatly appreciated.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**